app-shell/src/protocol-storage/file-system.ts

```diff
@@ -159,7 +165,12 @@ export function analyzeProtocolByKey(
     PROTOCOL_ANALYSIS_DIRECTORY_NAME
   )
   const destFilePath = makeAnalysisFilePath(analysisDirPath)
-  return analyzeProtocolSource(srcDirPath, destFilePath)
+  return readFilesWithinDirectory(srcDirPath).then(protocolFiles => {
+    if (protocolFiles.length === 0) {
+      throw new Error('No valid protocol files found')
+    }
+    return analyzeProtocolSource(protocolFiles[0], destFilePath)
```

Comment:
```suggestion
    return analyzeProtocolSource(dirContainingProtocol, destFilePath)
```
Let's ensure we only have one dir, too. See other comment.